Traceback (most recent call last):
  File "/Users/slimhy/anaconda3/lib/python3.10/site-packages/jupyter_cache/executors/utils.py", line 58, in single_nb_execution
    executenb(
  File "/Users/slimhy/anaconda3/lib/python3.10/site-packages/nbclient/client.py", line 1305, in execute
    return NotebookClient(nb=nb, resources=resources, km=km, **kwargs).execute()
  File "/Users/slimhy/anaconda3/lib/python3.10/site-packages/jupyter_core/utils/__init__.py", line 166, in wrapped
    return loop.run_until_complete(inner)
  File "/Users/slimhy/anaconda3/lib/python3.10/asyncio/base_events.py", line 649, in run_until_complete
    return future.result()
  File "/Users/slimhy/anaconda3/lib/python3.10/site-packages/nbclient/client.py", line 705, in async_execute
    await self.async_execute_cell(
  File "/Users/slimhy/anaconda3/lib/python3.10/site-packages/nbclient/client.py", line 1058, in async_execute_cell
    await self._check_raise_for_error(cell, cell_index, exec_reply)
  File "/Users/slimhy/anaconda3/lib/python3.10/site-packages/nbclient/client.py", line 914, in _check_raise_for_error
    raise CellExecutionError.from_cell_and_msg(cell, exec_reply_content)
nbclient.exceptions.CellExecutionError: An error occurred while executing the following cell:
------------------
"""
Base class for loading preprocessed 3D point clouds.

Args:
----
    ...:    See CompatLoader3D.
"""
from compat3D_PC import EvalLoader_PC

test_dataset = EvalLoader_PC(root_dir=ROOT_DIR,
                             semantic_level="coarse",
                             num_points=2048)

# Iterating over the dataset
for k in range(len(test_dataset)):
    shape_id, style_id, points = test_dataset[k]
    # Separate points and colors
    points_xyz = points[:, :3]
    points_col = points[:, 3:]

    # Print shape ID
    print("Shape ID: ", shape_id)
    print("Style ID: ", style_id)
    break

plt_utils.plot_pointclouds([np.array(points_xyz)],
                            colors=[np.array(points_col)],
                            size=15,
                            cmap='viridis',
                            point_size=8)
------------------


[0;31m---------------------------------------------------------------------------[0m
[0;31mFileNotFoundError[0m                         Traceback (most recent call last)
Cell [0;32mIn[7], line 10[0m
[1;32m      1[0m [38;5;124;03m"""[39;00m
[1;32m      2[0m [38;5;124;03mBase class for loading preprocessed 3D point clouds.[39;00m
[1;32m      3[0m 
[0;32m   (...)[0m
[1;32m      6[0m [38;5;124;03m    ...:    See CompatLoader3D.[39;00m
[1;32m      7[0m [38;5;124;03m"""[39;00m
[1;32m      8[0m [38;5;28;01mfrom[39;00m [38;5;21;01mcompat3D_PC[39;00m [38;5;28;01mimport[39;00m EvalLoader_PC
[0;32m---> 10[0m test_dataset [38;5;241m=[39m [43mEvalLoader_PC[49m[43m([49m[43mroot_dir[49m[38;5;241;43m=[39;49m[43mROOT_DIR[49m[43m,[49m
[1;32m     11[0m [43m                             [49m[43msemantic_level[49m[38;5;241;43m=[39;49m[38;5;124;43m"[39;49m[38;5;124;43mcoarse[39;49m[38;5;124;43m"[39;49m[43m,[49m
[1;32m     12[0m [43m                             [49m[43mnum_points[49m[38;5;241;43m=[39;49m[38;5;241;43m2048[39;49m[43m)[49m
[1;32m     14[0m [38;5;66;03m# Iterating over the dataset[39;00m
[1;32m     15[0m [38;5;28;01mfor[39;00m k [38;5;129;01min[39;00m [38;5;28mrange[39m([38;5;28mlen[39m(test_dataset)):

File [0;32m~/Documents/3DCoMPaT/doc/compat3D_PC.py:211[0m, in [0;36mEvalLoader_PC.__init__[0;34m(self, *args, **kwargs)[0m
[1;32m    210[0m [38;5;28;01mdef[39;00m [38;5;21m__init__[39m([38;5;28mself[39m, [38;5;241m*[39margs, [38;5;241m*[39m[38;5;241m*[39mkwargs):
[0;32m--> 211[0m     [38;5;28;43msuper[39;49m[43m([49m[43m)[49m[38;5;241;43m.[39;49m[38;5;21;43m__init__[39;49m[43m([49m[38;5;241;43m*[39;49m[43margs[49m[43m,[49m[43m [49m[43msplit[49m[38;5;241;43m=[39;49m[38;5;124;43m"[39;49m[38;5;124;43mtest[39;49m[38;5;124;43m"[39;49m[43m,[49m[43m [49m[38;5;241;43m*[39;49m[38;5;241;43m*[39;49m[43mkwargs[49m[43m)[49m

File [0;32m~/Documents/3DCoMPaT/doc/compat3D_PC.py:103[0m, in [0;36mCompatLoader3D_PC.__init__[0;34m(self, root_dir, split, semantic_level, num_points, transform, half_precision, normalize_points, is_rgb)[0m
[1;32m    101[0m [38;5;28mself[39m[38;5;241m.[39mis_test [38;5;241m=[39m split [38;5;241m==[39m [38;5;124m"[39m[38;5;124mtest[39m[38;5;124m"[39m
[1;32m    102[0m [38;5;28mself[39m[38;5;241m.[39mis_rgb [38;5;241m=[39m is_rgb
[0;32m--> 103[0m [38;5;28mself[39m[38;5;241m.[39mhdf5_data [38;5;241m=[39m [43mload_data[49m[43m([49m
[1;32m    104[0m [43m    [49m[43mhdf5_path[49m[43m,[49m
[1;32m    105[0m [43m    [49m[43mhalf_precision[49m[43m,[49m
[1;32m    106[0m [43m    [49m[43mnormalize_points[49m[38;5;241;43m=[39;49m[43mnormalize_points[49m[43m,[49m
[1;32m    107[0m [43m    [49m[43mis_test[49m[38;5;241;43m=[39;49m[38;5;28;43mself[39;49m[38;5;241;43m.[39;49m[43mis_test[49m[43m,[49m
[1;32m    108[0m [43m    [49m[43mis_rgb[49m[38;5;241;43m=[39;49m[43mis_rgb[49m[43m,[49m
[1;32m    109[0m [43m[49m[43m)[49m
[1;32m    110[0m [38;5;28mself[39m[38;5;241m.[39mnum_points [38;5;241m=[39m num_points
[1;32m    111[0m [38;5;28mself[39m[38;5;241m.[39mpoints [38;5;241m=[39m [38;5;28mself[39m[38;5;241m.[39mhdf5_data[[38;5;241m0[39m]

File [0;32m~/Documents/3DCoMPaT/doc/compat3D_PC.py:34[0m, in [0;36mload_data[0;34m(hdf5_path, half_precision, normalize_points, is_test, is_rgb)[0m
[1;32m     31[0m     [38;5;28;01mraise[39;00m [38;5;167;01mNotImplementedError[39;00m([38;5;124m"[39m[38;5;124mOnly RGB is supported for now.[39m[38;5;124m"[39m)
[1;32m     33[0m [38;5;66;03m# Loading HDF5 arrays in CPU RAM[39;00m
[0;32m---> 34[0m [38;5;28;01mwith[39;00m [43mh5py[49m[38;5;241;43m.[39;49m[43mFile[49m[43m([49m[43mhdf5_path[49m[43m,[49m[43m [49m[38;5;124;43m"[39;49m[38;5;124;43mr[39;49m[38;5;124;43m"[39;49m[43m)[49m [38;5;28;01mas[39;00m hdf5_f:
[1;32m     35[0m     points [38;5;241m=[39m np[38;5;241m.[39marray(hdf5_f[[38;5;124m"[39m[38;5;124mpoints[39m[38;5;124m"[39m][:])[38;5;241m.[39mastype(
[1;32m     36[0m         [38;5;124m"[39m[38;5;124mfloat32[39m[38;5;124m"[39m [38;5;28;01mif[39;00m [38;5;129;01mnot[39;00m half_precision [38;5;28;01melse[39;00m [38;5;124m"[39m[38;5;124mfloat16[39m[38;5;124m"[39m
[1;32m     37[0m     )
[1;32m     38[0m     shape_ids [38;5;241m=[39m np[38;5;241m.[39marray(hdf5_f[[38;5;124m"[39m[38;5;124mshape_id[39m[38;5;124m"[39m])

File [0;32m/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/h5py/_hl/files.py:567[0m, in [0;36mFile.__init__[0;34m(self, name, mode, driver, libver, userblock_size, swmr, rdcc_nslots, rdcc_nbytes, rdcc_w0, track_order, fs_strategy, fs_persist, fs_threshold, fs_page_size, page_buf_size, min_meta_keep, min_raw_keep, locking, alignment_threshold, alignment_interval, meta_block_size, **kwds)[0m
[1;32m    558[0m     fapl [38;5;241m=[39m make_fapl(driver, libver, rdcc_nslots, rdcc_nbytes, rdcc_w0,
[1;32m    559[0m                      locking, page_buf_size, min_meta_keep, min_raw_keep,
[1;32m    560[0m                      alignment_threshold[38;5;241m=[39malignment_threshold,
[1;32m    561[0m                      alignment_interval[38;5;241m=[39malignment_interval,
[1;32m    562[0m                      meta_block_size[38;5;241m=[39mmeta_block_size,
[1;32m    563[0m                      [38;5;241m*[39m[38;5;241m*[39mkwds)
[1;32m    564[0m     fcpl [38;5;241m=[39m make_fcpl(track_order[38;5;241m=[39mtrack_order, fs_strategy[38;5;241m=[39mfs_strategy,
[1;32m    565[0m                      fs_persist[38;5;241m=[39mfs_persist, fs_threshold[38;5;241m=[39mfs_threshold,
[1;32m    566[0m                      fs_page_size[38;5;241m=[39mfs_page_size)
[0;32m--> 567[0m     fid [38;5;241m=[39m [43mmake_fid[49m[43m([49m[43mname[49m[43m,[49m[43m [49m[43mmode[49m[43m,[49m[43m [49m[43muserblock_size[49m[43m,[49m[43m [49m[43mfapl[49m[43m,[49m[43m [49m[43mfcpl[49m[43m,[49m[43m [49m[43mswmr[49m[38;5;241;43m=[39;49m[43mswmr[49m[43m)[49m
[1;32m    569[0m [38;5;28;01mif[39;00m [38;5;28misinstance[39m(libver, [38;5;28mtuple[39m):
[1;32m    570[0m     [38;5;28mself[39m[38;5;241m.[39m_libver [38;5;241m=[39m libver

File [0;32m/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/h5py/_hl/files.py:231[0m, in [0;36mmake_fid[0;34m(name, mode, userblock_size, fapl, fcpl, swmr)[0m
[1;32m    229[0m     [38;5;28;01mif[39;00m swmr [38;5;129;01mand[39;00m swmr_support:
[1;32m    230[0m         flags [38;5;241m|[39m[38;5;241m=[39m h5f[38;5;241m.[39mACC_SWMR_READ
[0;32m--> 231[0m     fid [38;5;241m=[39m [43mh5f[49m[38;5;241;43m.[39;49m[43mopen[49m[43m([49m[43mname[49m[43m,[49m[43m [49m[43mflags[49m[43m,[49m[43m [49m[43mfapl[49m[38;5;241;43m=[39;49m[43mfapl[49m[43m)[49m
[1;32m    232[0m [38;5;28;01melif[39;00m mode [38;5;241m==[39m [38;5;124m'[39m[38;5;124mr+[39m[38;5;124m'[39m:
[1;32m    233[0m     fid [38;5;241m=[39m h5f[38;5;241m.[39mopen(name, h5f[38;5;241m.[39mACC_RDWR, fapl[38;5;241m=[39mfapl)

File [0;32mh5py/_objects.pyx:54[0m, in [0;36mh5py._objects.with_phil.wrapper[0;34m()[0m

File [0;32mh5py/_objects.pyx:55[0m, in [0;36mh5py._objects.with_phil.wrapper[0;34m()[0m

File [0;32mh5py/h5f.pyx:106[0m, in [0;36mh5py.h5f.open[0;34m()[0m

[0;31mFileNotFoundError[0m: [Errno 2] Unable to open file (unable to open file: name = './3D_PC_data/test_coarse.hdf5', errno = 2, error message = 'No such file or directory', flags = 0, o_flags = 0)

